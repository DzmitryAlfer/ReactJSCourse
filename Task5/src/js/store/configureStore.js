import { createStore, applyMiddleware, compose } from 'redux'
import { createLogger } from 'redux-logger'
import createSagaMiddleware, { END } from 'redux-saga'
import rootReducer from '../reducers'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'

const persistConfig = {
    key: 'root',
    storage,
};

const persistedReducer = persistReducer(persistConfig, rootReducer)

export default function configureStore(initialState) {
    const sagaMiddleware = createSagaMiddleware()

    const store = createStore(
        persistedReducer,
        initialState,
        compose(
            applyMiddleware(
                sagaMiddleware,
                createLogger()
            )
        )
    );
    const persistor = persistStore(store);

    /*if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept('../reducers', () => {
            const nextRootReducer = require('../reducers').default;
            store.replaceReducer(nextRootReducer);
        });
    }*/

    store.runSaga = sagaMiddleware.run;
    store.close = () => store.dispatch(END);
    return {store, persistor};
}