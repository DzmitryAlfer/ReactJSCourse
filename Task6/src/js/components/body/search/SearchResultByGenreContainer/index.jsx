import React, {PureComponent} from 'react';
import * as actions from "../../../../actions";
import {MovieList} from "../MovieList";
import {connect} from "react-redux";

class SearchResultByGenreContainer extends PureComponent {

    componentDidMount() {
        this.props.dispatch(actions.movies.fetchRelatedMovies(this.props.movie))
    }

    render() {
        return (<div className='search-results'>
                <MovieList movies={this.props.relatedMovies}/>
            </div>
        );
    }
}

const mapStateToProps = state => {
    const { movieReducer } = state;

    return {
        movie : movieReducer.movie,
        relatedMovies: movieReducer.relatedMovies
    }
};

export default connect (mapStateToProps)(SearchResultByGenreContainer);