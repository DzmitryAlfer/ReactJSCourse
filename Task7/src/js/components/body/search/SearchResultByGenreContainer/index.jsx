import React, {PureComponent} from 'react';
import * as actions from "../../../../actions";
import {MovieList} from "../MovieList";
import {connect} from "react-redux";

class SearchResultByGenreContainer extends PureComponent {

    componentDidMount() {
        if(this.props.movie){
            this.props.dispatch(actions.movies.fetchRelatedMovies(this.props.movie.genres));
        }
    }

    render() {
        return (<div className='search-results'>
                <MovieList movies={this.props.relatedMovies.data}/>
            </div>
        );
    }
}

const mapStateToProps = (state, ownProps) => {
    const { movieReducer } = state;

    return {
        movie : ownProps.movie,
        relatedMovies: movieReducer.relatedMovies
    }
};

export default connect (mapStateToProps)(SearchResultByGenreContainer);