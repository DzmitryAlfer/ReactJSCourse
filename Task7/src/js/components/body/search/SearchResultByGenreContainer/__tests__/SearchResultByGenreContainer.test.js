import React from 'react';
import Renderer from 'react-test-renderer';
import SearchResultByGenreContainer from '../index';
import { Provider } from 'react-redux';
import configureStore from '../../../../../store/configureStore';

const { store } = configureStore();

jest.mock('../../../../../services/api');

describe('SearchResultByGenreContainer', () => {
    test('Component render', () => {
        const tree = Renderer.create(<Provider store={store}>
            <SearchResultByGenreContainer movie={{
                id: 962,
                title: 'The Gold Rush',
                tagline: "The World's Greatest Laughing Picture!",
                vote_average: 7.8,
                vote_count: 411,
                release_date: '1925-06-25',
                poster_path: 'https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg',
                overview: 'A lone prospector ventures into Alaska looking for gold. He gets mixed up with some burly characters and falls in love with the beautiful Georgia. He tries to win her heart with his singular charm.',
                budget: 923000,
                revenue: 2500000,
                genres: ['Adventure', 'Comedy', 'Drama'],
                runtime: 95,
            }}
            />
        </Provider>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
