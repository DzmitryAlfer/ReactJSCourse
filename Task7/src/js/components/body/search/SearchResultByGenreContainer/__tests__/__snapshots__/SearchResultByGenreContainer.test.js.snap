// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultByGenreContainer Component render 1`] = `
<div
  className="search-results"
/>
`;
